armi/reactor/assemblies.py

```diff
@@ -180,6 +180,11 @@ def add(self, obj: blocks.Block):
             are updated. The axial mesh and other Block geometry parameters are
             updated in ``calculateZCoords``.
         """
+        if self.blockType is not None and not isinstance(obj, self.blockType):
+            msg = f"Cannot add {obj} to this Assembly, it is not a {self.blockType}."
+            runLog.error(msg)
+            raise TypeError(msg)
```

Comment:
Combine this logic with the logic in `self.insert` into a method? `self._checkPotentialChild`?

Non-blocking request

Reply:
Done.